pyanaconda/ui/tui/tuiobject.py

```diff
@@ -55,3 +86,223 @@ def has_help(self):
+        `add_to_container()` method.
+
+        Also set user prompt to Anaconda's default.
+
+        :param title: Name of the item which user is setting.
+        :type title: str
+
+        :param value: Actual value of the title. User will see what is set right now.
+        :type value: str
+
+        :param conditions: Optional acceptance conditions. If condition is not valid the
+                          `wrong_input_message` is printed and user must set correct value.
+        :type conditions: A function func(user_input, report_func) -> bool taking user input and returning bool.
+                          See `report_func` parameter for report_func specification.
+
+        :param report_func: Function for printing errors and warnings from conditions.
+        :type report_func: Function func(message)  -- taking one argument message.
+        """
+        super().__init__()
+        self._title = title
+        self._value = value
+        self._conditions = conditions
+        self._report_func = report_func
+
+        self._user_prompt = _("Enter a new value for '%(title)s' and press %(enter)s") % {
+                                # TRANSLATORS: 'title' as a title of the entry
+                                "title": title,
+                                # TRANSLATORS: 'enter' as the key ENTER
+                                "enter": Prompt.ENTER
+                               }
+        self._no_separator = False
+
+    @property
+    def title(self):
+        """Title of the item we want to get user input for.
+
+        :returns: Name of the item.
+        :rtype: str.
+        """
+        return self._title
+
+    @title.setter
+    def title(self, title):
+        """Set title of the item we want to get user input for.
+
+        :param title: Item title.
+        :type title: str.
+        """
+        self._title = title
+
+    @property
+    def value(self):
+        """Value of the item we want to set.
+
+        Optional.
+
+        :returns: Value user set or was predefined.
+        :rtype: str. Default: None
+        """
+        return self._value
+
+    @value.setter
+    def value(self, value):
+        """Set value of the item we want to set.
+
+        Optional.
+
+        :param value: Value of the item.
+        :type value: str.
+        """
+        self._value = value
+
+    @property
+    def no_separator(self):
+        """Print separator or hide user input as a actual screen part?
+
+        :returns: False if separator should be printed. True if not.
+        :rtype: bool. Default: False
+        """
+        return self._no_separator
+
+    @no_separator.setter
+    def no_separator(self, no_separator):
+        """Should the separator be printed?
+
+        :param no_separator: True to disable separator, False otherwise.
+        :type no_separator: bool.
+        """
+        self._no_separator = no_separator
+
+    def widget_name(self):
```

Comment:
I would like to change this method to the `widget` property, that will return the instance of `TextWidget`. It can simplify adding dialogs to the containers.

Another solution could be to create a class `DialogWidget` or `EntryWidget`, that would get a dialog and create a message for the container. But I think that the first solution might be better.

Reply:
I would go for both, for returning widget and returning only text. The point is, it is harder to change content of `TextWidget` if need to.